src/app/core/site-settings/site-settings.service.ts

```diff
@@ -58,6 +61,26 @@ export class SiteSettingsService extends LatestEntityLoader<SiteSettings> {
 
     this.initFromLocalStorage();
     this.cacheInLocalStorage();
+
+    //Listen to SiteSettings updates and sync the default language in localStorage
+    this.entityMapper
+      .receiveUpdates(SiteSettings)
+      .pipe(
+        untilDestroyed(this),
+        filter((update) => update?.entity.getId() === SiteSettings.ENTITY_ID),
+      )
+      .subscribe((updatedSiteSettings) => {
+        const updatedLanguage =
+          updatedSiteSettings?.entity?.defaultLanguage?.id;
+        const currentLanguage = localStorage.getItem(
+          LANGUAGE_LOCAL_STORAGE_KEY,
+        );
+        if (updatedLanguage !== currentLanguage) {
+          // Override the language in localStorage and reload the app
+          localStorage.setItem(LANGUAGE_LOCAL_STORAGE_KEY, updatedLanguage);
+          this.window.location.reload();
+        }
```

Comment:
could you extract this into a method in LanguageService and reuse the code for this as well as language-select component?

Reply:
I tried that, but it’s creating a circular dependency. If I add a new method in LanguageService and call it here, it causes a loop because LanguageService also uses SiteSettingsService. But I’ll check again to see if there’s a way to make it work.